### ISSUE TITLE
Remove minimum interval value

### ISSUE THREAD

**Issue by melvinlemoine** [louislam/uptime-kuma#1645]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

Remove the interval of 20 seconds minimum from heartbeat interval & heartbeat retry interval.

### ✔️ Solution

Remove the interval of 20 seconds minimum from heartbeat interval & heartbeat retry interval. I need to monitore under 20 seconds and each time I need to save my 10s seconds interval & 5s retry interval I need to remove min="20" from HTML.

**Comment by ylluminate** [comment]
This is a very important feature when this is being used for gateway and ISP monitoring. This should be able to go down to at least 1 second intervals.

**Comment by yurividal** [comment]
+1 for this, especially for the PING sensors.

**Comment by Computroniks** [comment]
> +1. I don't see the reason for a minimum when it's self-hosted.

I don't know, but I think that it was probably added to prevent poor performance with lots of monitors. Having said that, no reason why we couldn't just add a warning saying as much if the user drops it below 20

**Comment by Laukage** [comment]
Hope this gets added ^^

**Comment by akinnee** [comment]
> > +1. I don't see the reason for a minimum when it's self-hosted.
> 
> I don't know, but I think that it was probably added to prevent poor performance with lots of monitors. Having said that, no reason why we couldn't just add a warning saying as much if the user drops it below 20

Yep. I'd say allow us to shoot ourselves in the foot, but warn that we might do so. 😄

**Comment by Laukage** [comment]
> > > +1. I don't see the reason for a minimum when it's self-hosted.
> > 
> > I don't know, but I think that it was probably added to prevent poor performance with lots of monitors. Having said that, no reason why we couldn't just add a warning saying as much if the user drops it below 20
> 
> Yep. I'd say allow us to shoot ourselves in the foot, but warn that we might do so. 😄 

How so? Many ping tools has 1sec intervals

**Comment by Computroniks** [comment]
It has been implemented in #1740 and is awaiting merge

**Comment by jwklijnsma** [comment]
we will i be mirg ?

**Comment by KRider92** [comment]
Bumping this... @louislam is there any chance this will be merged soon?

**Comment by Computroniks** [comment]
> Bumping this... @louislam is there any chance this will be merged soon?

I need to implement some of the changes identified in the PR before it can be merged, most notably requiring the user to confirm that they actually want to go below 20 seconds. More info can be found in the #1740 thread. I will probably get around to implementing this in the next 1 or two weeks

**Comment by MostHated** [comment]
Any movement planned on this? This is definitely a much needed change. 20 seconds is basically an eternity for important/critical/actionable info.

**Comment by CommanderStorm** [comment]
Please refer to what was previously posted

> More info can be found in the #1740 thread

TLDR: As long as the performance issues are not resolved, this will not get merged. There are a few performance optimisations implemented in `1.23` and some in `2.0`
⇒ let's see if this addresses the issue enough to make this a good change from a support perspective. Adding this without fixing the underlying performance issues is not a thing we can do.

**Comment by yurividal** [comment]
> Please refer to what was previously posted
> 
> > More info can be found in the #1740 thread
> 
> TLDR: As long as the performance issues are not resolved, this will not get merged. There are a few performance optimisations implemented in `1.23` and some in `2.0` ⇒ let's see if this addresses the issue enough to make this a good change from a support perspective. Adding this without fixing the underlying performance issues is not a thing we can do.

I haven't looked at the code (and I am, by no means, a professional developer), but, one thing I know is that apps like telegraf, and PRTG, which allow for hundreds of ping sensors with minimal intervals is that they seem to spin a separate thread for every ping. Do you know if this is the same approach Kuma is taking?

**Comment by Computroniks** [comment]
> I haven't looked at the code (and I am, by no means, a professional developer), but, one thing I know is that apps like telegraf, and PRTG, which allow for hundreds of ping sensors with minimal intervals is that they seem to spin a separate thread for every ping. Do you know if this is the same approach Kuma is taking?

If I remember correctly, the issue is not with the actual running of the tests, but with the backend database, particularly the calculation for the uptime requiring very large queries which end up overloading the database. More info is in #2750.

**Comment by Axsgranted** [comment]
I too am looking for less than 20 second intervals but I'm guessing this merge is dead now? I mean after 2 years...

**Comment by CommanderStorm** [comment]
In v1 the dangers of such an option are too high (v1 can only cope with a db of 1..10GB depending on storage and retention).
From a support perspective I can't in good faith support merging this.

See comment from above:

> As long as the performance issues are not resolved, this will not get merged. There are a few performance optimisations implemented in 1.23 and some in 2.0
⇒ let's see if this addresses the issue enough to make this a good change from a support perspective. Adding this without fixing the underlying performance issues is not a thing we can do.

=> v2.0 is tracked in https://github.com/louislam/uptime-kuma/issues/4500 but has some bugs that need adressing before we can release the first beta. If a benchmark gets made we and said benchmark proves that this PR is possible, we can merge it (removing the warning)

**Comment by xvertile** [comment]
I have created a fork of the latest version where the minimum interval is disabled you can find it here:

https://github.com/xvertile/uptime-kuma

Cheers

**Comment by AhmadMuj** [comment]
Can we have any update on this please ?

**Comment by CommanderStorm** [comment]
The situation has not changed from my last post https://github.com/louislam/uptime-kuma/issues/1645#issuecomment-2080472341.
We/someone from the community have not made a benchmark to see if 20x-ing the load is acceptable.

Why we are so cautions with this is explained by the fact that 500 monitors @ 20s is equivalent to 25 monitors @ 1s..
v2.0 should better for this usecase, but it might not be enough...

**Comment by Axsgranted** [comment]
Could you not simply have a warning when setting it lower than 20seconds?

**Comment by CommanderStorm** [comment]
See https://github.com/louislam/uptime-kuma/issues/1645#issuecomment-2080472341

> From a support perspective I can't in good faith support merging this. I am struggling to respond to users/reviewing PRs as is.

**Comment by oskerlau** [comment]
> I have created a fork of the latest version where the minimum interval is disabled you can find it here:
> 
> https://github.com/xvertile/uptime-kuma
> 
> Cheers

Great !! Is there a docker image?

**Comment by Kristoffeh** [comment]
> Great !! Is there a docker image?

Did you get this tested?

**Comment by crisog** [comment]
I had been following this and the possibility of having less than 20s pings, in this issue https://github.com/louislam/uptime-kuma/issues/5057#issuecomment-2314121572 the main bottleneck was V1.

I just saw [2.0.0-beta.2](https://github.com/louislam/uptime-kuma/releases/tag/2.0.0-beta.2) released and I'm wondering how close it is to being able to reduce this to a lower number (maybe not 1 sec, but lower)

**Comment by CommanderStorm** [comment]
@Kristoffeh Nobody has benchmarked this to our knowlege

**Comment by CommanderStorm** [comment]
@crisog
If you're interested in helping out, benchmarking would be a great way to contribute!

As noted above, the current default is in place for untested monitor types to prevent an overwhelming number of issues for maintainers. As I mentioned earlier, we're already struggling with this, and increasing our workload further wouldn't be sustainable.

**Comment by Maingron** [comment]
I totally agree - I want to monitor if a local device is available or not. 

If you want to avoid issue flooding, maybe you could still decrease the minimum interval value for private address spaces. Pinging private addresses should be pretty reliable unless they're down.

**Comment by CommanderStorm** [comment]
@Maingron This seems entirely off-topic. Maybe you posted in the wrong window?

**Comment by vitich** [comment]
Any news? 20 seconds is too much on self-hosted... Forced to look for alternatives.

**Comment by xvertile** [comment]
I have run monitors at 5 seconds for weeks it went fine?

**Comment by CommanderStorm** [comment]
How many monitors? Reducing this value is blocked on the benchmarking if reducing this value is actually Ok.

**Comment by MrNickIE** [comment]
Hello, I see this is marked as completed.

I am considering moving from Smokeping to Uptime Kuma, but this was a showstopper for us.  Is this now available?

**Comment by CommanderStorm** [comment]
It was merged into 2.1 and will be available once that release is made.
You can subscribe to our release notifications on the front page.

I don't know if we have done a nightly for this.
